pkg/controllers/jobset_controller.go

```diff
@@ -139,6 +139,13 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	// Handle suspending a jobset or resuming a suspended jobset.
 	jobsetSuspended := js.Spec.Suspend != nil && *js.Spec.Suspend
+	// Get the latest from jobset as updating status for jobs seems
+	// to cause a cache modified error
+	if err := r.Get(ctx, req.NamespacedName, &js); err != nil {
+		// we'll ignore not-found errors, since there is nothing we can do here.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
```

Comment:
That is the not the pattern, in each "Reconcile" we work with the copy of jobset that triggered the update. 

Also, even if we get here, an update could also happen right after, so there is really no escape from reconciling again.

Reply:
This certainly will not help, and reconciling on failure is a feature not a bug.

As I commented below, the issue with the tests is that the test entries are using the same jobset at the same time, we need to make sure that each test entry starts fresh.